### PR TITLE
[Feature] Drop down icons

### PR DESCRIPTION
**Pull request by Civolilah**
Here are two screenshots of implemented solution for dropdown icons:

Screenshot 2022-12-28 at 17 50 28

Screenshot 2022-12-28 at 17 50 41

@beganovich @turbo124 Now all dropdown menus have icons implemented. In addition to working on adding icons, I refactored some small parts of the code to follow the current convention and create better formatting (specially related to dropdown menus). Let me know your thoughts.

### CODE REVIEW

**Comment by turbo124** [comment]
LGTM if it is ok with @beganovich

File: src/components/icons/Icon.tsx
Comment (beganovich) [inline src/components/icons/Icon.tsx:10]: I don't think I understand what's the purpose of this component. Why can't we just directly pass the icon reference to the dropdown component instead of having deeply locked Icon component?

File: src/components/icons/Icon.tsx
Comment (beganovich) [inline src/components/icons/Icon.tsx:26]: This is not the correct naming scheme. The correct prop should be `size` since you are in the context of icons.

File: src/components/icons/Icon.tsx
Comment (beganovich) [inline src/components/icons/Icon.tsx:33]: Not following accent color.

File: src/pages/clients/common/components/CustomResourcefulActions.tsx
Comment (beganovich) [inline src/pages/clients/common/components/CustomResourcefulActions.tsx:49]: I don't think I get this. What does icon type "new" has to do with "View statement"?

**Review by beganovich — CHANGES_REQUESTED** [review]
Looks good, few changes are needed.

1. Icons in the dropdown are not following the accent color schemes.
2. Colorful buttons for bulk actions may look cool in blue color, but as soon as you switch to something else 🤮

File: src/components/dropdown/DropdownElement.tsx
Comment (Civolilah) [reply src/components/dropdown/DropdownElement.tsx:20]: Actually, apart from this one `iconType` we have an `icon` prop, the `iconType` prop is only used for frequent icons (New, Archive...), but another icon prop allows us to pass maybe a specific icon.

File: src/components/dropdown/DropdownElement.tsx
Comment (Civolilah) [reply src/components/dropdown/DropdownElement.tsx:37]: `{props.iconType && !props.icon && <Icon type={props.iconType} />}`

Icons from the `Icon` component will only appear if the `icon` prop doesn't exist, so we won't have a case where we have two icons.

File: src/components/dropdown/DropdownElement.tsx
Comment (Civolilah) [reply src/components/dropdown/DropdownElement.tsx:20]: > Actually, apart from this one `iconType` we have an `icon` prop, the `iconType` prop is only used for frequent icons (New, Archive...), but another icon prop allows us to pass maybe a specific icon. But, as we discussed on the Slack the logic will be slightly changed.

File: src/components/icons/Icon.tsx
Comment (Civolilah) [reply src/components/icons/Icon.tsx:10]: As we discussed on the Slack this component will just get new shape of purpose.

File: src/pages/clients/common/components/CustomResourcefulActions.tsx
Comment (Civolilah) [reply src/pages/clients/common/components/CustomResourcefulActions.tsx:49]: Hmm, I think that you missed it. It's actually, 'view'.

File: src/components/icons/Icon.tsx
Comment (Civolilah) [reply src/components/icons/Icon.tsx:26]: Changed

File: src/components/icons/Icon.tsx
Comment (Civolilah) [reply src/components/icons/Icon.tsx]: Now, we are using accentColor

File: src/components/dropdown/Dropdown.tsx
Comment (Civolilah) [inline src/components/dropdown/Dropdown.tsx:40]: The reason for the style change for this component is the wider width of all the dropdown menus when we included the icons at the beginning. Tailwind doesn't give us a `min-w-48` (for example) class to set, so I had to set it through the style property.

**Comment by Civolilah** [comment]
@beganovich @turbo124 The icons are assigned to all the drop-down menus I know. Please let me know if I've missed any dropdowns in the app.

**Comment by turbo124** [comment]
@Civolilah 

Looks good, however we need to make the edit action the first option across the application.

**Comment by Civolilah** [comment]
> @Civolilah
> 
> Looks good, however we need to make the edit action the first option across the application.

@turbo124 It's implemented and as we discussed on Slack, I've removed the entity name from the dropdowns where appropriate as it's obvious that editing that entity. (`edit_product` -> `edit`) Also, description of this PR is updated with screenshots of how pages look now.

**Comment by turbo124** [comment]
@Civolilah @beganovich 

LGTm

File: src/components/dropdown/Dropdown.tsx
Comment (beganovich) [reply src/components/dropdown/Dropdown.tsx:40]: This is okay, but Tailwind min/max are defined in REMs. Can we do the same? Like this (I think) it's in pixels which means it's not fully reactive to different screen sizes?

**Comment by beganovich** [comment]
Looks good, just to resolve that issue with defining the width and height in rems.

File: src/components/dropdown/Dropdown.tsx
Comment (Civolilah) [reply src/components/dropdown/Dropdown.tsx:40]: Changed

**Comment by Civolilah** [comment]
> Looks good, just to resolve that issue with defining the width and height in rems.

@beganovich done

**Comment by beganovich** [comment]
Hey, @Civolilah looks like we have some merge conflicts. Can you fix them so I can merge?

**Comment by Civolilah** [comment]
> 

@beganovich Of course, I'll move this PR to draft until I resolve the minor merge conflicts.

**Comment by Civolilah** [comment]
@beganovich Merge conflicts resolved.

**Comment by beganovich** [comment]
Looks good, thanks :+1:

**Pull request merged**
